src/pint/models/timing_model.py

```diff
@@ -1502,6 +1496,57 @@ def delete_jump_and_flags(self, toa_table, jump_num):
             return
         self.components["PhaseJump"].setup()
 
+    def get_spin_freq(self, times, calctype="modelF0"):
+        """Return barycentric pulsar rotational frequency in Hz at specific times
+
+        Parameters
+        ----------
+        times : float or long double MJD (can be array), astropy.Time object, TOAs
+            The times (barycentric, inf freq) at which to compute the spin frequency
+        calctype : {'modelF0', 'numerical', 'taylor'}
+            Type of calculation:
+                "modelF0" means a constant ``F0`` from the timing model
+                "numerical" uses the numerical derivative ``d_phase_d_toa()``
+                    This option requires times to be a TOAs object
+                "taylor" uses a Taylor series to compute the frequencies
+
+        Returns
+        -------
+        freq : astropy.units.Quantity (can be array)
+            The spin frequency in Hz at each barycentric time
+
+        Note
+        ----
+        "numerical" is significantly slower, but is much more exact, and
+        """
+        calc = calctype.lower()
+        assert calc in ["modelf0", "taylor", "numerical"]
+
+        if calc == "modelf0":
+            return self.F0.quantity
+
+        # ToDo: How does this handle Glitch or Piecewise models?
+        if calc == "numerical":
+            assert isinstance(times, TOAs)
+            return self.d_phase_d_toa(times)
+
+        # Handle various types of input times for "taylor" calc
+        if isinstance(times, TOAs):
+            # If we pass the TOA table, then barycenter the TOAs
+            bts = self.get_barycentric_toas(times)  # have units of days
```

Comment:
Do we actually want to barycenter the TOAs (like, make this dependent on sky position)? If so, should there be a faster shortcut computation method (`fasttaylor`) that just uses the un-barycentered times?

Reply:
Yeah, I don't know the answer to the first question.  I think it depends on your use case and how precise you want things.  A `fasttaylor` method is probably a good idea no matter what, though.